scheduler/scheduler.go

```diff
@@ -813,13 +775,7 @@ func (s *scheduler) HandleGomitEvent(e gomit.Event) {
 		}).Debug("event received")
 		// We need to unsubscribe from deps when a task goes disabled
 		task, _ := s.getTask(v.TaskID)
-		depGroups := getWorkflowPlugins(task.workflow.processNodes, task.workflow.publishNodes, task.workflow.metrics)
-		for k := range depGroups {
-			mgr, err := task.RemoteManagers.Get(k)
-			if err == nil {
-				mgr.UnsubscribeDeps(task.ID())
-			}
-		}
```

Comment:
**To reviewers**:
This code from in lines 816-822 occurred in few places, so I decided to put that in [(t *task) UnsubscribePlugins()](https://github.com/intelsdi-x/snap/pull/1548/files#diff-77f163e381dd113e90ebd1bd76479b54R364)